zcash_test_vectors/zip_0032.py

```diff
@@ -26,23 +26,28 @@ def MKGh(Context, IKM):
     I_R = I[32:]
     return (I_L, I_R)
 
-def CKDh(Context, sk_par, c_par, i):
+def CKDh(Context, sk_par, c_par, i, tag, full_width_leaf):
     assert type(Context) == HardenedOnlyContext
     assert 0x80000000 <= i and i <= 0xFFFFFFFF
+    assert type(tag) == bytes
+    assert type(full_width_leaf) == bool
 
-    I   = prf_expand(c_par, Context.CKDDomain + sk_par + i2leosp(32, i))
+    leaf = b"" if tag == b"" and not full_width_leaf else int(full_width_leaf).to_bytes()
+    I = prf_expand(c_par, Context.CKDDomain + sk_par + i2leosp(32, i) + leaf + tag)
     I_L = I[:32]
     I_R = I[32:]
     return (I_L, I_R)
 
-class ArbitraryKey(object):
-    Arbitrary = HardenedOnlyContext(b'ZcashArbitraryKD', b'\xAB')
 
-    def __init__(self, IKM, path, sk, chaincode):
+class RegisteredKey(object):
+    Context = HardenedOnlyContext(b'ZIPRegistered_KD', b'\xAC')
```

Comment:
After the reversion, this would then become (for spec closeness):
```suggestion
    Registered = HardenedOnlyContext(b'ZIPRegistered_KD', b'\xAC')
```